client/repolist.c

```diff
@@ -111,6 +134,7 @@ TDNFLoadRepoData(
         closedir(pDir);
     }
     TDNF_SAFE_FREE_MEMORY(pszRepoFilePath);
+    TDNF_SAFE_FREE_STRINGARRAY(ppszUrlIdTuple);
```

Comment:
why is this needed? if ppszUrlIdTuple[1] is null and ppszUrlIdTuple[0] has garbage address, it might crash.